assets/js/modules/search-console/components/module/ModulePopularPagesWidget/index.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * ModulePopularPagesWidget component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { numFmt } from '../../../../../util';
+import { MODULES_SEARCH_CONSOLE, STORE_NAME, DATE_RANGE_OFFSET } from '../../../datastore/constants';
+import PreviewTable from '../../../../../components/PreviewTable';
+import Link from '../../../../../components/Link';
+import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import TableOverflowContainer from '../../../../../components/TableOverflowContainer';
+import ReportTable from '../../../../../components/ReportTable';
+import { isZeroReport } from '../../../util/is-zero-report';
+import { generateDateRangeArgs } from '../../../util/report-date-range-args';
+import Header from './Header';
+
+const { useSelect } = Data;
+
+function ModulePopularPagesWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
+	const {
+		data,
+		isLoading,
+		error,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
+		const { startDate, endDate } = dateRangeDates;
+
+		const reportArgs = {
+			startDate,
+			endDate,
+			url: url || undefined,
+			dimensions: 'query',
+			limit: 10,
+		};
+
+		return {
+			data: store.getReport( reportArgs ),
+			isLoading: ! store.hasFinishedResolution( 'getReport', [ reportArgs ] ),
+			error: store.getErrorForSelector( 'getReport', [ reportArgs ] ),
+		};
+	} );
+
+	if ( isLoading ) {
+		return <PreviewTable padding />;
```

Comment:
It would be nice for the loading state here to still return it wrapped in the regular `Widget` including the `Header` (and `Footer`), so that the loading UI is displayed in the body of the regular widget and the header and CTA are already there. This is similar to how the new Analytics `ModuleOverviewWidget` does it.